src/Loop/Driver.php

```diff
@@ -285,4 +311,39 @@
      * @return null|object|resource The loop handle the event loop operates on. `null` if there is none.
      */
     abstract public function getHandle();
+
+    /**
+     * Create a watcher identifier.
+     *
+     * Any watcher identifiers returned to clients by the driver MUST be created using this method. The returned watcher
+     * identifier SHALL be unique for the duration of the PHP process.
+     *
+     * @return string The watcher identifier.
+     */
+    final protected function createWatcherId()
+    {
+        return $this->watcherIdPrefix . $this->nextWatcherId++;
+    }
+
+    /**
+     * Validate a watcher identifier which has been passed to the driver by a client.
+     *
+     * The driver MUST call this method when it is provided with a watcher identifier which it does not recognise. The
+     * driver SHOULD NOT call this method every time it receives a watcher identifier, as doing so would be relatively
+     * expensive and is not typically necessary.
+     *
+     * @param string $watcherId The watcher identifier.
+     *
+     * @throws InvalidWatcherException If the watcher identifier does not belong to this driver.
+     */
+    final protected function validateWatcherId($watcherId)
```

Comment:
Given that it doesn't return a boolean I'd rather call it `assertOwnWatcherId` or something like that.